hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HStore.java

```diff
@@ -254,48 +244,47 @@
   protected HStore(final HRegion region, final ColumnFamilyDescriptor family,
       final Configuration confParam, boolean warmup) throws IOException {
 
-    this.fs = region.getRegionFileSystem();
-
-    // Assemble the store's home directory and Ensure it exists.
-    fs.createStoreDir(family.getNameAsString());
-    this.region = region;
-    this.family = family;
     // 'conf' renamed to 'confParam' b/c we use this.conf in the constructor
     // CompoundConfiguration will look for keys in reverse order of addition, so we'd
     // add global config first, then table and cf overrides, then cf metadata.
     this.conf = new CompoundConfiguration()
-      .add(confParam)
-      .addBytesMap(region.getTableDescriptor().getValues())
-      .addStringMap(family.getConfiguration())
-      .addBytesMap(family.getValues());
-    this.blocksize = family.getBlocksize();
+        .add(confParam)
+        .addBytesMap(region.getTableDescriptor().getValues())
+        .addStringMap(family.getConfiguration())
+        .addBytesMap(family.getValues());
+
+    this.region = region;
+    this.storeContext = initializeStoreContext(family);
+
+    // Assemble the store's home directory and Ensure it exists.
+    getRegionFileSystem().createStoreDir(getColumnFamilyName());
+
+    this.blocksize = getColumnFamilyDescriptor().getBlocksize();
 
     // set block storage policy for store directory
-    String policyName = family.getStoragePolicy();
+    String policyName = getColumnFamilyDescriptor().getStoragePolicy();
```

Comment:
nit: why bother with this change at all? 'family' is passed on the constructor. Its final. Why bother going via the accessor in the constructor?

Reply:
fixed.